src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/HdcedCalculator.kt

```diff
@@ -176,19 +293,39 @@ class HdcedCalculator(
   private fun getDeductedDays(sentenceCalculation: SentenceCalculation) =
     sentenceCalculation.adjustments.deductions
 
-  private fun calculateHdcedMinimumCustodialPeriod(
+  private fun calculateHdcedMinimumCustodialPeriodUsingPreHdc365Rules(
     sentence: CalculableSentence,
     sentenceCalculation: SentenceCalculation,
     parentRule: CalculationRule,
     params: HdcedParams,
   ) {
-    sentenceCalculation.homeDetentionCurfewEligibilityDate =
-      sentence.sentencedAt.plusDays(hdcedConfiguration.minimumDaysOnHdc).plusDays(
-        params.addedDays,
+    sentenceCalculation.hdcedByCalcType[InterimHdcCalcType.HDCED_PRE_365_RULES] = sentence.sentencedAt.plusDays(hdcedConfiguration.minimumDaysOnHdc).plusDays(params.addedDays)
+    sentenceCalculation.noDaysToHdcedByCalcType[InterimHdcCalcType.HDCED_PRE_365_RULES] = hdcedConfiguration.minimumDaysOnHdc.plus(params.addedDays)
+    sentenceCalculation.breakdownByInterimHdcCalcType[InterimHdcCalcType.HDCED_PRE_365_RULES] =
+      ReleaseDateCalculationBreakdown(
+        rules = setOf(CalculationRule.HDCED_MINIMUM_CUSTODIAL_PERIOD, parentRule),
+        rulesWithExtraAdjustments = mapOf(
+          CalculationRule.HDCED_MINIMUM_CUSTODIAL_PERIOD to AdjustmentDuration(
+            hdcedConfiguration.minimumDaysOnHdc,
+          ),
+        ),
+        adjustedDays = params.addedDays,
+        releaseDate = sentenceCalculation.hdcedByCalcType[InterimHdcCalcType.HDCED_PRE_365_RULES]!!,
+        unadjustedDate = sentence.sentencedAt,
       )
-    sentenceCalculation.numberOfDaysToHomeDetentionCurfewEligibilityDate =
-      hdcedConfiguration.minimumDaysOnHdc.plus(params.addedDays)
-    sentenceCalculation.breakdownByReleaseDateType[ReleaseDateType.HDCED] =
+  }
+
+  // This is a copy of calculateHdcedMinimumCustodialPeriodUsingPreHdc365Rules - the difference being it saves against the new Post HDC-365 variant variables and associated config
+  // TODO potential to combine the methods into one, will look into separately
+  private fun calculateHdcedMinimumCustodialUsingPostHdc365Rules(
```

Comment:
I'm good with this for now - but there is lots of scope for us to make this much more elegant :)